### DISCUSSION TITLE
Low speed (WAN upload / download speed) data not displaying well on Statistics Graph Card?

### DISCUSSION THREAD

**Discussion by rholligan** [Vaskivskyi/ha-asusrouter#852] [General]
I recently set up this integration to pull data from my Asus XT-8 into Home Assistant. Specifically I am interested in the WAN upload / download speed. Prior to this, I was monitoring the same data using the [UPnP/IGD](https://www.home-assistant.io/integrations/upnp) integration.

When comparing the two datasets, I notice that the UPNP integration seems to better display data on the Statistics Graph Card when speeds are low (in the range 0.010 to 10 Mbit/s). Conversely the Asus Router integration seems to better pick up maximum speeds. 

See attached screenshots comparing the same time period for the two integrations. Checking the values being logged at a specific time (06:00) shows that both integration log pretty much the same values for min / mean / max, yet this doesn't seem to translate well to the Statistics Graph Card.

My initial guess is that it's to do with the range on the y-axis of the chart. Because Asus Router is logging higher maximum values, the y-axis is being adjusted to show better that top end. 

Any ideas why this is happening? Are there any settings I can adjust to avoid losing visibility of this low speed data with Asus Router integration?

**Asus Router**
![image](https://github.com/user-attachments/assets/358ca3b4-3e25-4780-af2c-983ad6c85a18)
![image](https://github.com/user-attachments/assets/a9d54209-8040-48aa-81e3-38f7ed0f2391)

**UPNP**
![image](https://github.com/user-attachments/assets/2f3b9a6f-650c-41ee-964b-d0aa2e389150)
![image](https://github.com/user-attachments/assets/0dcb5a28-63ae-4c20-a23a-f2c3fc0eb6bf)

**Comment by Vaskivskyi** [comment]
Unfortunately, I don't know how exactly UPnP integration works. But this would be my guess on why the data is different between the two.

AsusRouter is averaging the speed over the period between the sensor updates. It does not even get the speed - just calculates it from the network traffic in 2 points and divides it by the time between the points.

This means, that if your sensors are updated e.g. once per 30 seconds (default) and you download 300 Mb/s for 5 of them and don't do anything for the rest 25, the speed you get will be ~ 50 Mb/s.

So, from the integration point of view - you can adjust the time between sensor updates - this will change the data you are saving in the entities. (`Configure - Update Intervals - Entities update` -> this will change intervals for all the entities). Alternatively, you can enable per-sensor update interval (`Operation mode - Checkbox on the bottom`). Then just change `Network stat data` in `Update intervals`.

You might want to try different values which would effectively change what you are seeing. Going higher with time means you'll avoid hight peaks. This can be both good or bad depending on your use case.
